tests/SbrpTests/ValidationTests.cs

```diff
@@ -0,0 +1,178 @@
+            Directory.GetDirectories(Path.Combine(Config.RepoRoot, "src/targetPacks/ILsrc"))
+                .Union(Directory.GetDirectories(Path.Combine(Config.RepoRoot, "src/textOnlyPackages/src")))
+                .Select(x => Path.GetFileName(x).ToLower())
+        );
+
+        var filteredPackages = packages
+            .Where(package =>
+            {
+                string packageName = Path.GetFileNameWithoutExtension(package).ToLower();
+                packageName = Regex.Replace(packageName, VersionPattern, string.Empty);
+                return !targetAndTextOnlyPacks.Contains(packageName);
+            });
+
+        Output.WriteLine($"Checking {filteredPackages.Count()} packages for SBRP attribute.");
+
+        foreach (var package in filteredPackages)
+        {
+            string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName(), Path.GetFileNameWithoutExtension(package));
+
+            try
+            {
+                Directory.CreateDirectory(tempDirectory);
+                ZipFile.ExtractToDirectory(package, tempDirectory);
+                var dlls = Directory.GetFiles(tempDirectory, "*.dll", SearchOption.AllDirectories);
+
+                foreach (var dll in dlls)
+                {
+                    using FileStream stream = new (dll, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                    using PEReader peReader = new (stream);
+                    MetadataReader reader = peReader.GetMetadataReader();
+
+                    Assert.True(HasSbrpAttribute(reader), $"{package}/{Path.GetRelativePath(tempDirectory, dll)} does not contain the {SbrpAttributeType} attribute with key='source' and value='source-build-reference-packages'.");
+                }
+            }
+            finally
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+    }
+
+    [SkippableFact]
+    public async Task ValidateSignatures()
+    {
+        Skip.If(RuntimeInformation.IsOSPlatform(OSPlatform.Windows), "This test is not supported on Windows.");
+
+        string[] packages = GetPackages();
+
+        ISignatureVerificationProvider[] trustProviders = [new SignatureTrustAndValidityVerificationProvider()];
+        PackageSignatureVerifier verifier = new (trustProviders);
+        var settings = SignedPackageVerifierSettings.GetDefault();
+
+        Output.WriteLine($"Checking {packages.Count()} packages for signatures.");
+
+        foreach (var package in packages)
+        {
+            bool isSigned = await IsPackageSignedAsync(package, verifier, settings);
+            Assert.False(isSigned, $"{package} is signed. Signed packages are not allowed in source-build-reference-packages.");
+        }
+    }
+
+    private string[] GetPackages()
+    {
+        string buildPackagesDirectory = Path.Combine(Config.RepoRoot, "artifacts/source-build/self/src/artifacts/packages", Config.BuildType, "Shipping");
```

Comment:
I'm not a fan of hardcoding this path. This would break whenever we change the build layout (i.e. when removing the source build inner clone). We should use a runtimeconfiguration value for that instead so that it stays in sync with the msbuild build time property.

Reply:
I'm open to creating a new environment variable that defaults to this path. That said, it'd still have to be hardcoded somewhere. Given the scale of this repo, I'm comfortable leaving this as is.